ArtOfIllusion/src/artofillusion/texture/ProceduralTextureCommon.java

```diff
@@ -0,0 +1,163 @@
+/* Copyright (C) 2017 by Maksim Khramov
+
+   This program is free software; you can redistribute it and/or modify it under the
+   terms of the GNU General Public License as published by the Free Software
+   Foundation; either version 2 of the License, or (at your option) any later version.
+   This program is distributed in the hope that it will be useful, but WITHOUT ANY 
+   WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A 
+   PARTICULAR PURPOSE.  See the GNU General Public License for more details. */
+
+package artofillusion.texture;
+
+import artofillusion.MaterialPreviewer;
+import artofillusion.TextureParameter;
+import artofillusion.image.ImageMap;
+import artofillusion.math.RGBColor;
+import artofillusion.procedural.IOPort;
+import artofillusion.procedural.ImageModule;
+import artofillusion.procedural.Module;
+import artofillusion.procedural.OutputModule;
+import artofillusion.procedural.ParameterModule;
+import artofillusion.procedural.Procedure;
+import artofillusion.procedural.ProcedureEditor;
+import static artofillusion.texture.Texture.*;
+import artofillusion.ui.ActionProcessor;
+import artofillusion.ui.Translate;
+import artofillusion.ui.ValueSelector;
+import buoy.event.ValueChangedEvent;
+import buoy.widget.BDialog;
+import buoy.widget.BorderContainer;
+import buoy.widget.LayoutInfo;
+import buoy.widget.RowContainer;
+import java.awt.Rectangle;
+
+/**
+ *
+ * @author MaksK
+ */
+public class ProceduralTextureCommon
+{
+  private ProceduralTextureCommon()
+  {    
+  }
+  
+  /**
+   * Create a Procedure object for texture.
+   */
+
+  public static Procedure createProcedure()
```

Comment:
This method is the odd one out, as it is specific to _Texture_ procedures. Perhaps call it `createTextureProceedure()`? Perhaps it does not really belong here, despite the fact that two types of texture use the same code.